### ISSUE TITLE
Shiny Proxy cannot be used directly inside a Docker container

### ISSUE THREAD

**Issue by Ununnilium** [openanalytics/shinyproxy#60]
There is a configuration option `internal-networking` which allows running Shiny Proxy inside a Docker container:
```
shiny:
  proxy:
    docker:
      internal-networking: true
```
This option does not work directly. There are the following problems:
- The app containers are created without linking them to the Shiny Proxy container and without specifying a Docker network. Therefore, Shiny Proxy is not able to connected to the app containers. When no network is specified, Docker connects the new container to the default bridge network. This is not recommended anymore. The bridge network has no name resolution, one has to use the `link` option instead. I think it would be better to start the app containers in a special network so the name resolution works without problems.
Solution at the moment: Run in background a script which connects the Shiny Proxy and the app containers to a manually created Docker network.

- Shiny Proxy does not support connecting to Docker with a Unix domain socket anymore, although in the past it has worked. Therefore, one can not just mount the host Docker socket `/var/run/docker.sock` as a volume and use it directly.
Solution at the moment: Either activate the TCP socket (no so easy to configure secure) or use socat inside the container:
```
socat TCP-LISTEN:2375,reuseaddr,fork,bind=127.0.0.1 UNIX-CONNECT:/var/run/docker.sock
```

I have written a Python script which circumvents the problems:
https://github.com/Ununnilium/shinyproxy-docker

Although the script works good, I think it would be good to fix the problems directly in Shiny Proxy.

**Comment by tverbeke** [comment]
Hi @Ununnilium, 

Thanks for the feedback. When you say the option does not work, did you follow the step by step guide at https://github.com/openanalytics/shinyproxy-config-examples/tree/master/02-containerized-docker-engine ? Should additional information be added there?

**Comment by Ununnilium** [comment]
Hi @tverbeke,
Thank you for the fast answer. I'm sorry for the bug report. I didn't saw this very nice documentation and didn't set the `docker-network` options for the apps. I found now also the reason for the problem with the Unix socket: in the past, I had to set `shiny` - `proxy` - `docker` - `url` to `unix:///var/run/docker.sock`, now it's enough to just not set this option, it will not work anymore when it's set to `unix:///var/run/docker.sock` or `unix://var/run/docker.sock`.
